### PR TITLE
[Darktrace] Initial Release for the Darktrace

### PR DESCRIPTION
**Pull request by vinit-elastic**
## What does this PR do?

- Generated the skeleton of the Darktrace integration package.
- Added data streams.
- Added data collection logic for all the data streams.
- Added the ingest pipeline for all the data streams.
- Mapped fields according to the ECS schema and added Fields metadata in the appropriate yml files.
- Added dashboards and visualizations.
- Added test for pipeline for all the data streams.
- Added system test cases for all the data streams.

## Integration release checklist
This checklist is intended for integrations maintainers to ensure consistency
when creating or updating a Package, Module or Dataset for an Integration.

### All changes
- [x] Change follows the [contributing guidelines](https://github.com/elastic/integrations/blob/master/CONTRIBUTING.md)
- [x] Supported versions of the monitoring target are documented
- [x] Supported operating systems are documented (if applicable)
- [x] Integration or [System tests](https://github.com/elastic/elastic-package/blob/master/docs/howto/system_testing.md) exist
- [x] Documentation exists
- [x] Fields follow [ECS](https://github.com/elastic/ecs) and [naming conventions](https://www.elastic.co/guide/en/beats/devguide/master/event-conventions.html)
- [x] At least a manual test with ES / Kibana / Agent has been performed.
- [x] Required Kibana version set to: ^8.2.1

### New Package
- [x] Screenshot of the "Add Integration" page on Fleet added

### Dashboards changes
- [x] Dashboards exists
- [x] Screenshots added or updated
- [x] Datastream filters added to visualizations

### Log dataset changes
- [x] [Pipeline tests](https://github.com/elastic/elastic-package/blob/master/docs/howto/pipeline_testing.md) exist (if applicable)
- [x] Generated output for at least 1 log file exists
- [x] Sample event (sample_event.json) exists

## How to test this PR locally

- Clone integrations repo.
- Install elastic package locally.
- Start elastic stack using elastic-package.
- Move to integrations/packages/darktrace directory.
- Run the following command to run tests. 
> elastic-package test 

## Related issues

- Closes https://github.com/elastic/integrations/issues/2754

## Screenshots

image
image
image
image
image

### CODE REVIEW

**Review by efd6 — APPROVED** [review]
Query: There are a lot of object arrays in the documents produced here under the darktrace key path. Is this a concern?

**Comment by vinit-elastic** [comment]
Hey @efd6, Yes the Darktrace API sends the data in such a way. And We can't flatten in as it would result in conflicts in index mappings. So that doesn't seem to be a good idea to flatten it.

**Comment by efd6** [comment]
Thanks

**Comment by jamiehynds** [comment]
@andrewkroh @efd6 could we prioritise review on this integration next please? Thanks!

File: packages/darktrace/_dev/deploy/docker/files/config.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+rules:
+  - path: /modelbreaches
+    methods: ["GET"]
```
Comment (P1llus) [inline packages/darktrace/_dev/deploy/docker/files/config.yml:3]: Does this API use any sort of authentication? Any way we could add that in as well?

File: packages/darktrace/data_stream/ai_analyst_alert/agent/stream/httpjson.yml.hbs [hunk 2]
```diff
@@ -0,0 +1,48 @@
+config_version: 2
+interval: {{interval}}
+request.timeout: 5m
```
Comment (P1llus) [inline packages/darktrace/data_stream/ai_analyst_alert/agent/stream/httpjson.yml.hbs:3]: Could this be a configurable advanced option which defaults to 5min? We usually allow users to override this in some niche usecases.

File: packages/darktrace/data_stream/model_breach_alert/agent/stream/httpjson.yml.hbs [hunk 3]
```diff
@@ -0,0 +1,51 @@
+config_version: 2
+interval: {{interval}}
+request.timeout: 5m
```
Comment (P1llus) [inline packages/darktrace/data_stream/model_breach_alert/agent/stream/httpjson.yml.hbs:3]: Same as above, please make this configurable

**Review by P1llus — APPROVED** [review]
I am a bit concerned about the amount of arrays in the resulting data, but I don't feel there is any good way around it, since creating single documents for each would simply be too many.

I presume there has not been any big challenges in still using them for visualizations?

I am also a bit unsure how this impacts possibility to create SIEM rules for certain content, seeing as they are all stored in arrays.

If this is something that has already been discussed/thought about, feel free to ignore that.

Outside of that I added a few small comments, the rest seems good to go, great work! :)

**Comment by P1llus** [comment]
I think maybe we have to discuss the duplicate custom fields, as has been brought up in some other PR comments as well

**Pull request merged**